libs/shared-entity/src/dto/workspace_dto.rs

```diff
@@ -1,8 +1,9 @@
 use chrono::{DateTime, Utc};
 use collab_entity::{CollabType, EncodedCollab};
+use collab_folder::{Folder, ViewIcon};
```

Comment:
shouldn't reference to `collab_folder` here.

Reply:
I can move the logic of building the folder view. But for `ViewIcon`(required for web display), I'll have to duplicate from the `collab-folder`.

```rust
#[derive(Eq, PartialEq, Debug, Hash, Clone, Serialize_repr, Deserialize_repr)]
#[repr(u8)]
pub enum IconType {
  Emoji = 0,
  Url = 1,
  Icon = 2,
}

#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
pub struct ViewIcon {
  pub ty: IconType,
  pub value: String,
}
```

Reply:
move `collab_folder_to_folder_view` to `get_user_workspace_structure` then you don't need to depend on collab-folder

Reply:
removed `collab_folder` dep on `shared-entitiy`